--- main1.c	2025-12-21 22:15:35
+++ main2.c	2025-12-21 22:16:09
@@ -3,7 +3,7 @@
 #include <time.h>
 #include <string.h>
 
-#define N (argc > 1 ? atoi(argv[1]) : 6)
+#define N (argc > 2 ? atoi(argv[2]) : 6)
 
 void gen(int h, int w, char maze[h][w]) {
     for (int i = 0; i < h; i++) 
@@ -53,6 +53,13 @@
 
 int main(int argc, char *argv[]) {
     srand(time(NULL));
+    
+    char pass = '.', wall = '#';
+    if (argc > 1 && strlen(argv[1]) >= 2) {
+        pass = argv[1][0];
+        wall = argv[1][1];
+    }
+    
     int h = 2*N+1, w = 2*N+1;
     char maze[h][w];
     
@@ -60,7 +67,7 @@
     
     for (int i = 0; i < h; i++) {
         for (int j = 0; j < w; j++) 
-            putchar(maze[i][j]);
+            putchar(maze[i][j] == '.' ? pass : wall);
         putchar('\n');
     }
     return 0;
